here-naksha-lib-jbon/src/commonMain/kotlin/JbPath.kt

```diff
@@ -8,20 +8,93 @@ import kotlin.js.JsExport
-            //       Map the view into a default reader
-            //       Split the path
-            //       Go along the path through the reader and find the target
-            //       Return the value if of the type; otherwise alternative
+        var bytes: ByteArray = ByteArray(0)
+        lateinit var view: IDataView
+        var reader: JbReader = JbReader()
+        var jmap: JbMap = JbMap()
+        var feature: JbFeature = JbFeature().mapReader(reader)
+
+        fun getBool(binary: ByteArray, path: String, alternative: Boolean? = null): Boolean? {
+            val valueReader = readElement(binary, path)
+            if (valueReader != null && valueReader.isBool()) {
+                return valueReader.readBoolean() ?: alternative
+            }
+            return alternative
+        }
+
+        fun getInt32(binary: ByteArray, path: String, alternative: Int? = null): Int? {
+            val valueReader = readElement(binary, path)
+            if (valueReader != null && valueReader.isInt32()) {
+                return valueReader.readInt32()
+            }
+            return alternative
+        }
+
+        fun getInt64(binary: ByteArray, path: String, alternative: Int? = null): Int? {
+            TODO("int64 has to be implemented")
+        }
+
+        fun getFloat32(binary: ByteArray, path: String, alternative: Float? = null): Float? {
+            val valueReader = readElement(binary, path)
+            if (valueReader != null && valueReader.isFloat32()) {
+                return valueReader.readFloat32()
+            }
+            return alternative
+        }
+
+        fun getDouble(binary: ByteArray, path: String, alternative: Double? = null): Double? {
+            val valueReader = readElement(binary, path)
+            if (valueReader != null) {
+                if (valueReader.isFloat32()) {
+                    return valueReader.readFloat32().toDouble()
+                } else if (valueReader.isFloat64()) {
+                    return valueReader.readDouble()
+                }
+            }
+            return alternative
+        }
+
+        fun getString(binary: ByteArray, path: String, alternative: String? = null): String? {
+            val valueReader = readElement(binary, path)
+            if (valueReader != null && valueReader.isString()) {
+                return valueReader.readString()
+            }
             return alternative
         }
+
+        private fun readElement(binary: ByteArray, path: String): JbReader? {
+            val session = JbSession.get()
+            if (!bytes.contentEquals(binary)) {
+                bytes = binary
+                view = session.newDataView(bytes)
+                reader = JbReader()
+                reader.mapView(view, 0)
+                feature = JbFeature().mapReader(reader)
+                jmap = JbMap()
+                jmap.mapReader(feature.reader)
+            }
+
+            if (feature.reader.isMap()) {
```

Comment:
if json is array we return alternative immediately, later we can add support for querying such paths
`properties.links[0].id`